LibGAP.jl/src/gap.jl

```diff
@@ -37,16 +38,37 @@ Base.setproperty!(x::MPtr, f::Symbol, v) = GAP.GAPFuncs.ASS_REC(x, RNamObj(f), t
 
 import Base: *, +, -, /, ^, mod, <, ==
 
-#
-+(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.SUM(x, y)
--(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.DIFF(x, y)
-*(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.PROD(x, y)
-/(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.QUO(x, y)
-^(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.POW(x, y)
-mod(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.MOD(x, y)
+typecombinations = [[:GAPInputType_noint,:GAPInputType_noint],
+                    [:GAPInputType_noint,:Int64],
+                    [:Int64,:GAPInputType_noint]]
+function_combinations = [[:(+),:SUM],
+                         [:(-),:DIFF],
+                         [:(*),:PROD],
+                         [:(/),:QUO],
+                         [:(^),:POW],
+                         [:(mod),:MOD],
+                         [:(<),:LT],
+                         [:(==),:EQ]]
 
-<(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.LT(x, y)
-==(x::GAPInputType, y::GAPInputType) = GAP.GAPFuncs.EQ(x, y)
+for types in typecombinations
+    for func in function_combinations
+        @eval begin
+            $(func[1])(x::$(types[1]),y::$(types[2])) = GAP.GAPFuncs.$(func[2])(x,y)
+        end
+    end
+end
```

Comment:
Sehr schön!

In der Zukunft können wir natürlich noch viele mehr machen, z. B. `lcm`, `gcd`, `rem`, ... Aber da sollte man dann evtl. überlegen, ob das so sinnvoll ist.

Frage: wenn man `<` und `==` definiert, stellt dann Julia automatisch `>`, `<=`, `>=` bereit? Falls nicht, sollte man das noch tun (kann man aus `LT` basteln durch tauschen der Eingaben bzw. `!` verwenden).

Reply:
Yes, Julia overloads it. But it still does not really work as intended, see
```
julia> x < 1
GAP: false

julia> x > 1
GAP: true
```
Maybe we should convert bools automatically?

Reply:
Are Julia true/false singletons? Then yes. Otherwise I'd just insert a call to from_gap_bool

Reply:
Ah, I see, there are `jl_true` and `jl_false`. So yes, I'd simply always auto-convert these between GAP and Julia.

Reply:
And sorry for my german comments, I had a braino there.

Reply:
No worries. I will autoconvert them.